File: Components/Dashboard/CS/DashboardMainDemo/Dashboards/SalesPerformance.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.DataAccess;
using DevExpress.DashboardCommon;

namespace DashboardMainDemo.Dashboards {
    public partial class SalesPerformance : Dashboard {
        public SalesPerformance() {
            InitializeComponent();
        }
        void SalesPerformance_DataLoading(object sender, DashboardDataLoadingEventArgs e) {
            SalesPerformanceDataGenerator dataGenerator = new SalesPerformanceDataGenerator(DataLoader.LoadSales());
            dataGenerator.Generate();
            if(e.DataSource == dsMonthlySales)
                e.Data = dataGenerator.MonthlySales;
            else if(e.DataSource == dsTotalSales)
                e.Data = dataGenerator.TotalSales;
            else if(e.DataSource == dsKeyMetrics)
                e.Data = dataGenerator.KeyMetrics;
        }
    }
}

File: Components/Dashboard/VB/DashboardMainDemo/Dashboards/SalesPerformance.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.DataAccess
Imports DevExpress.DashboardCommon

Namespace DashboardMainDemo.Dashboards
	Partial Public Class SalesPerformance
		Inherits Dashboard
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub SalesPerformance_DataLoading(ByVal sender As Object, ByVal e As DashboardDataLoadingEventArgs) Handles MyBase.DataLoading
			Dim dataGenerator As New SalesPerformanceDataGenerator(DataLoader.LoadSales())
			dataGenerator.Generate()
			If e.DataSource Is dsMonthlySales Then
				e.Data = dataGenerator.MonthlySales
			ElseIf e.DataSource Is dsTotalSales Then
				e.Data = dataGenerator.TotalSales
			ElseIf e.DataSource Is dsKeyMetrics Then
				e.Data = dataGenerator.KeyMetrics
			End If
		End Sub
	End Class
End Namespace
